addons/wiki/models.py

```diff
@@ -180,11 +180,12 @@ def save(self, *args, **kwargs):
     def check_spam(self):
         request, user_id = get_request_and_user_id()
         user = OSFUser.load(user_id)
+        if not user:  # for tests and admin operations
+            return False
+
         request_headers = string_type_request_headers(request)
         node = self.wiki_page.node
 
-        if not settings.SPAM_CHECK_ENABLED:
-            return False
```

Comment:
why not changing to check the new config value? Also, the logic in  Line 207 needs to be updated/relocated, since `do_spam_check` no longer returns anything meaningful